source/components/IDInput/index.jsx

```diff
@@ -34,12 +34,12 @@ const IDInput = ({
     () => [principalId, accountId].includes(value), [principalId, accountId, value],
   );
 
-  const isIncludesContacts = useMemo(() => !contacts
+  const notInContacts = useMemo(() => !contacts
     .flatMap((c) => c.contacts)
     .map((c) => c.id)
-    .includes(value), [contacts]);
+    .includes(value), [contacts, value]);
 
-  const shouldDisplayContacts = value !== null && isValid && isIncludesContacts && !isUserAddress;
+  const shouldDisplayContacts = value !== null && isValid && notInContacts && !isUserAddress;
```

Comment:
Should have a `!` here? Or we are changing the logic of when we should display contacts